compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Base.hs

```diff
@@ -107,6 +107,20 @@ newtype PartyLiteral = PartyLiteral{unPartyLiteral :: T.Text}
     deriving stock (Eq, Data, Generic, Ord, Show)
     deriving newtype (Hashable, NFData)
 
+-- | Human-readable name of a package. Must match the regex
+--
+-- > [a-zA-Z0-9_-]+
+newtype PackageName = PackageName{unPackageName :: T.Text}
+    deriving stock (Eq, Data, Generic, Ord, Show)
+    deriving newtype (Hashable, NFData)
+
+-- | Human-readable version of a package. Must match the regex
+--
+-- > [0-9]+(\.[0-9]+)*
+newtype PackageVersion = PackageVersion{unPackageVersion :: T.Text}
```

Comment:
We could make this something more structured but at least on the protobuf side, I’d like to keep it a text to make it easier to extend it to semver or whatever in the future and for now, keeping it a `Text` seems fine on the Haskell side (and we can change that whenever we want to).